src/cbapi/connection.py

```diff
@@ -94,8 +96,10 @@ def __init__(self, credentials, integration_name=None):
         if not credentials.ssl_verify_hostname:
             self.session.mount(self.server, HostNameIgnoringAdapter())
 
+        self._timeout = timeout
+
         # TODO: apply this to the ssl_verify_hostname case as well
-        self.session.mount(self.server, HTTPAdapter(max_retries=MAX_RETRIES))
+        self.session.mount(self.server, HTTPAdapter(max_retries=ZERO_RETRIES if self._timeout else MAX_RETRIES))
```

Comment:
I just did some experiments and you can keep the retry logic with the timeouts at the same time (tested with requests 2.9.1 on Python 2.7.10 Mac OS X). I'll merge this then revert back to `MAX_RETRIES`.